models/src/main/thrift/content/Platform.thrift

```diff
@@ -2,28 +2,52 @@ namespace scala com.gu.contentapi.client.model.v1
 
 include "CapiDateTime.thrift"
 
-struct Platforms {
-  0: optional WebPlatform web;
+enum Platform {
+  WEB = 0
+  PRINT = 1
+  EDITION = 2
+}
+
+struct PublicationInstance {
+  // let's have a platform enum to avoid ambiguity of this struct (only fields
+  // in the selected platform are relevant)
+  0: required Platform platform
+
+  1: optional WebFields webFields
 
-  1: optional PrintPlatform print;
+  2: optional PrintFields printFields
 
-  2: optional EditionPlatform edition;
+  3: optional EditionFields editionFields
```

Comment:
should we use the union type here since if something is in WEB platform it will never have PrintFields?

Reply:
(though caveat: unions can be a pain to work with in scala as we have to pattern match everywhere)

Reply:
Historically we've tended to avoid them for this reason as they make the SerDe code more complex.